bin/cqlsh.py

```diff
@@ -1497,7 +1497,7 @@ def describe_schema_client(self, include_system=False):
                 self.print_recreate_keyspace(k, sys.stdout)
                 print('')
 
-    def do_describe(self, parsed):
+    def do_describe(self, parsed, client_side=False):
```

Comment:
The name of the `client_side` parameter implies that it controls whether client-side describe will be used or not. It's not entirely true because in case of unsupported `client_side=False` we will fall back to client-side describe. How about `force_client_side_describe`?

Moreover, I'm not sure whether this should be a public parameter. The upstream doesn't allow to control whether client- or server-side describe is used. Perhaps we should turn this into a private function and keep a public `do_describe` without the additional parameter.

Reply:
I am not sure how this private function should work. Would it work to add a new parameter (force_client_side_describe = False) to the Shell class, set it to True before executing do_describe if the server doesn't support the DESCRIBE query, and then revert it back to False to attempt server-side by default?

Reply:
regardless of compatibility with upstream, which is already broken, since we introduce the client side back (I did it when we start working on this repo)

I think it would be nice to have an escape hatch for users, not sure how many would really need it.

Reply:
I was thinking about:

```python
def do_describe(self, parsed):
    self._do_describe(force_client_side_describe=False)

def _do_describe(self, force_client_side_describe):
    # ...
        except cassandra.protocol.SyntaxException:
            self._do_describe(force_client_side_describe=True)
```

Reply:
> I think it would be nice to have an escape hatch for users, not sure how many would really need it.

In that case, the transformation I suggested shouldn't be applied.

I still think that the parameter name `client_side` should be changed to `force_client_side_describe`, though.

Reply:
the suggestion would drift us more than the upstream, so yes it's not really that critical.
yes, the suggestion is a bit clearer, even that that not force involved here :), but any "bool" like prefix would work (use/is), and the suffix of describe, well.. that the name of the function, so might not be totally needed

Reply:
@piodul @fruch I'm not sure which version I should choose, the initial one or the one proposed by @piodul

Reply:
I think the current name is good enough